docker/Dockerfile

```diff
@@ -1,42 +1,118 @@
-## -*- docker-image-name: "triage-v4.0.0" -*-
+## -*- docker-image-name: "triage" -*-
 
-FROM python:3.7-alpine
+FROM python:3.7-alpine AS development
 
-LABEL triage.version="v4.0.0" \
-      triage.from="cli" \
-      creator="Center for Data Science and Public Policy (DSaPP)" \
-      maintainer="Adolfo De Unánue <adolfo@uchicago.edu>"
+LABEL creator="Center for Data Science and Public Policy (DSaPP)" \
+      maintainer="Adolfo De Unánue <adolfo@cmu.edu>"
 
-# the lapack package is only in the community repository
-RUN echo "http://dl-4.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories
+ENV PACKAGES="\
+    dumb-init \
+    musl \
+    libc6-compat \
+    linux-headers \
+    build-base \
+    bash \
+    git \
+    ca-certificates \
+    freetype \
+    freetype-dev \
+    libgfortran \
+    libgcc \
+    libstdc++ \
+    lapack-dev \
+    postgresql-dev \
+    libpng-dev \
+    openblas \
+    tcl \
+    tk \
+    libpq \
+    gcc \
+    g++ \
+    git \
+    curl \
+    libffi-dev \
+    jpeg-dev \ 
+    postgresql \
+    ncurses \
+    "
 
-RUN apk --update add --no-cache \
-        build-base \
-        lapack-dev \
-        gcc \
-        git \
-        postgresql-dev \
-        libpng-dev freetype-dev \
-        libpq
+RUN apk add --virtual build-runtime \
+     build-base python-dev openblas-dev lapack-dev freetype-dev pkgconfig gfortran g++ gcc git curl  \
+     && ln -s /usr/include/locale.h /usr/include/xlocale.h \
+     && pip install --upgrade pip setuptools \
+     && rm -r /root/.cache \
+        && apk add --no-cache --virtual build-dependencies $PACKAGES
 
+RUN mkdir triage
 
-RUN apk add --no-cache --virtual .build-deps \
-        lapack-dev \
-        gfortran \
-        musl-dev \
-        g++
+WORKDIR triage
 
-RUN ln -s /usr/include/locale.h /usr/include/xlocale.h
+ENV SHELL=/bin/bash
+ENV USERNAME=triage
+ENV USERID=1000
 
-COPY requirements.txt .
+RUN adduser \
+        --disabled-password \
+        --gecos "" \
+        --home "/home/triage" \
+        --uid "${USERID}" \
+        "${USERNAME}"         
 
-RUN pip install --no-cache-dir -r requirements.txt
+USER ${USERNAME}
 
-RUN mkdir triage
+RUN echo 'export PS1="\[$(tput setaf 4)$(tput bold)[\]triage@$(tput setaf 5)development$(tput setaf 4)$:\\w]#\[$(tput sgr0) \]"' >> /home/triage/.bashrc
 
-WORKDIR triage
+COPY requirements.txt /tmp
+
+RUN pip install -r /tmp/requirements.txt
+
+ENTRYPOINT [ "bash" ]
+
+FROM development AS from-master
+
+LABEL triage.version="master"
 
-# removing dependencies
-RUN apk del .build-deps
+USER root
+
+RUN pip install git+https://github.com/dssg/triage.git
+
+RUN apk del build-runtime \
+    && rm -rf /var/cache/apk/*
+
+RUN echo 'export PS1="\[$(tput setaf 4)$(tput bold)[\]triage@$(tput setaf 6)master$(tput setaf 4)$:\\w]#\[$(tput sgr0) \]"' > /home/triage/.bashrc
 
 ENTRYPOINT [ "triage" ]
+
+FROM from-master AS dirtyduck
+
+LABEL triage.version="stable"
```

Comment:
Isn't the triage version at this stage still from master -- it doesn't seem like it gets swapped out to the pypi version until the next stage of the build below, right?

Reply:
`dirtyduck` will use the version form master

Reply:
Just added a different `LABEL` in order to avoid confussion